Add entity to role using id (engine)

remcowesterhoud opened the pull request:
## Description

<!-- Describe the goal and purpose of this PR. -->

Refactors the engine to store entities by a role using the entity id and role id.

## Checklist

<!--- Please delete options that are not relevant. Boxes should be checked by reviewer. -->
- [ ] for CI changes:
  - [ ] structural/foundational changes signed off by [CI DRI](https://github.com/cmur2)
  - [ ] [ci.yml](https://github.com/camunda/camunda/blob/main/.github/workflows/ci.yml) modifications comply with ["Unified CI" requirements](https://github.com/camunda/camunda/wiki/CI-&-Automation#workflow-inclusion-criteria)
  - [ ] enable backports [when recommended](https://github.com/camunda/camunda/wiki/CI-&-Automation#when-to-backport-ci-changes)

## Related issues

closes #30116

remcowesterhoud committed:
feat: change error message from key to ID

remcowesterhoud committed:
feat: verify authorization before role existence

If a user is not authorized to access a role we don't have to check for the role's existence. This way we also don't let
 unauthorized users know if this role exists or not.

remcowesterhoud committed:
refactor: migrate role state to use entity id instead of key

The role state still stored entities by key. This needs to change to an id as all entities are now id-based.

remcowesterhoud committed:
refactor: add entity to role using id instead of keys

remcowesterhoud committed:
test: use proper role ids

The role ids were set to string values of the keys so the tests succeeded during the refactoring. We can now set these
to sensible values instead.

remcowesterhoud committed:
feat: add permissions to default role using id instead of key

remcowesterhoud committed:
feat: store role memberships using id instead of key

remcowesterhoud committed:
test: modify test cases to add entity using ids

remcowesterhoud committed:
feat: add mapping rules memberships to role using membership state

remcowesterhoud committed:
feat: retrieve role authorized resource identifiers using ids

remcowesterhoud committed:
test: assign entities to role using ids

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/camunda/camunda?pullRequest=30672) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/camunda/camunda?pullRequest=30672) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: remcowesterhoud<br/>:x: nicpuppa<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/camunda/camunda?pullRequest=30672) it.</sub>

remcowesterhoud committed:
feat: initialize identity using ids instead of keys

remcowesterhoud committed:
test: change tests to work with ids instead keys

remcowesterhoud committed:
fix: detect of mapping is already assigned to role

If a mapping is already assigned we should not add it to the role again.

remcowesterhoud committed:
test: ignore still unsupported cases

These cases will be recovered at different parts of the role refactoring.

remcowesterhoud committed:
test: refactor tests to use role id and entity id

I've ignored some tests and linked an issue number where they will be unignored.

remcowesterhoud committed:
test: disable IT until the exporter is refactored

This IT tries to fetch the data from ES, but for this we need to refactor the API and exporter to store and retrieve
roles using the role key

remcowesterhoud committed:
docs: adjust issue number

We cannot delete this property just yet.

nicpuppa committed:
fix: adjust after rebase

remcowesterhoud committed:
test: prevent NPE because of missing treepath

It doesn't matter which process gets started by this test. It worked with an incident before, but this has the chance of
 running into a NPE as the treepath could be null. By changing this to a service task proces we circumvent this problem.

remcowesterhoud committed:
fix: set role id in initializer

remcowesterhoud committed:
test: temporarily disable test

This test relies on authorization to be exported in a proper way. This will be solved in another issue.
